mungegithub/mungers/publish_scripts/util.sh

```diff
@@ -56,6 +61,18 @@ sync_repo() {
 
     git checkout ${currBranch}
 
+    # we must reset Godeps.json to what it looked like BEFORE the last vendor sync so that any
+    # new Godep.json changes from k8s.io/kubernetes will apply cleanly.  Since its always auto-generated
+    # it doesn't matter that we're removing it
+    lastResyncCommit=$(git rev-list -n 1 --grep "sync: resync vendor folder" HEAD)
+    cleanGodepJsonCommit=$(git rev-list -n 1 ${lastResyncCommit}^)
+    git checkout ${cleanGodepJsonCommit} Godeps/Godeps.json
+    if git diff --cached --exit-code &>/dev/null; then
+        echo "no need to reset Godeps.json!"
+    else
+        git -c user.name="Kubernetes Publisher" -c user.email="k8s-publish-robot@users.noreply.github.com" commit -m "sync: reset Godeps.json" -- Godeps/Godeps.json
+    fi
+
```

Comment:
I mostly agree if i understand you correctly. Are you saying we should check if the commits that are to be cherry-picked changes Godeps.json. If they do, we reset Godeps.json first, otherwise, we skip the reset?